domain/application/state/unit.go

```
@@ -1923,8 +1968,10 @@ WHERE     u.uuid = $unitUUID.uuid
 			return errors.Capture(err)
 		}
 		err := tx.Query(ctx, queryUnitPublicAddressesStmt, ident).GetAll(&address)
-		if err != nil {
-			return errors.Errorf("querying public addresses for unit %q: %w", uuid, err)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return nil
+		} else if err != nil {
+			return errors.Errorf("querying addresses for unit %q: %w", uuid, err)
```

**todo:** As Nico suggested, inline `err !=nil  && !errors.Is-err, sqlair.ErrNoRows)`

---

You don't need to do that. The first thing errors.Is does is check for nil. You're just adding boiler plate.

---

> You don't need to do that.

I think we do, my suggested change is to ignore the error *only* if it's `sqlair.ErrNoRows`, and the way to do it is with `err != nil && !errors.Is(err, sqlair.ErrNoRows)` right?

---

yes